activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -788,6 +831,12 @@ def try_to_checkout_new_connection
         def adopt_connection(conn)
           conn.pool = self
           @connections << conn
+
+          # We just created the first connection, it's time to load the schema
+          # cache if that wasn't eagerly done before
+          if @schema_cache.nil? && ActiveRecord.lazily_load_schema_cache
```

Comment:
@matthewd @jhawthorn do you still use `ActiveRecord.lazily_load_schema_cache` ? (https://github.com/rails/rails/pull/43372). 

Otherwise I think it'd be worth removing.

Reply:
We do have it turned on... but I'll try to figure out whether we still need it. I feel like the original motivation was around the fact that the eager loading didn't handle secondary connections (at least, in a way that worked for us at the time), and less about being affirmatively lazy.